todo/src/Components/TodoList.js

```diff
@@ -0,0 +1,78 @@
+import React, {Component} from 'react';
+import {connect} from 'react-redux';
+import {addTodo, toggleComplete, deleteTodo} from '../actions';
+
+import TodoItem from './TodoItem';
+import TodoForm from './TodoForm';
+
+class TodoList extends Component {
+    constructor() {
+        super();
+        this.state = {
+            newTodo: ''
+        }
+    }
+
+    handleChange = e => {
+        this.setState({[e.target.name]: e.target.value});
+    }
+
+    addTodo = e => {
+        e.preventDefault();
+        const newTodo = {
+            id: (this.props.todos.length - 1) + 1,
```

Comment:
Why do -1 + 1 here? You could just use `this.props.todos.length` for the same effect.